addons/project/static/src/views/project_project_controller_mixin.js

```diff
@@ -0,0 +1,18 @@
+import { user } from "@web/core/user";
+
+export const ProjectProjectControllerMixin = (T) =>
```

Comment:
I would simplify this by defining a function
```js
updateEmbeddedSettings(orm, actionId, resModel, resIds)
```
that can be called in your 3 views. No need for a complex inheritance pattern with a mixin imo.